### DISCUSSION TITLE
Timelapse view

### DISCUSSION THREAD

**Discussion by CameronFraser** [mkkellogg/GaussianSplats3D#280] [Ideas]
I have seen a few examples of timelapse support in splat viewers out in the wild specifically to support visualization of captures that happen with regular frequency such as construction. The pro version of the gauzilla editor: https://www.gauzilla.xyz/ and voluma.ai creating a timelapse mode: https://radiancefields.com/voluma-ai-adds-timelapse-mode

What do you think would be involved to add a feature like this? The initial challenges seem to be around loading and unloading efficiently while making the user experience feel seamless.

**Comment by mkkellogg** [comment]
If each frame (splat scene) of the time-lapse is small enough (in terms of splat count), I think this kind of functionality is achievable right now using one of a couple different approaches, both of which involve loading all splat scenes up front and then adjusting their visibility & opacity as appropriate as the animation progresses. The `Viewer` class has an accessor  function (`getSplatScene()`) to retrieve an individual splat scene and from there the splat scene's opacity and/or visibility can be set. Alternatively, I think you could just set up multiple instances of `DropInViewer`, one for each frame, at which point you could use the `visible` property of `DropInViewer` to set the visibility like any three.js object (although I think I would need to modify `SplatMaterial` so that the opacity could be set in a similar manner).

The key to this kind of approach would be managing memory usage. So in order to support a large number of frames, the size of each splat scene would need to be small. If only a small number of frames is required, that would give you more flexibility in per-scene splat count. For the Voluma.ai demo, I can see in the browser's network tab that 6 small splat scenes are being downloaded. One is a base scene, and the others appear to be cropped scenes that only contain data that changes between frames. All of the scenes are quite small; the base scene is 17 MB and the rest are each less than 10 MB. 

The `Viewer` class does have functions for adding and removing splat scenes at run-time, but they are not designed for seamless operation and the viewer will definitely become unresponsive for a brief time when they are called. For now, I would avoid those if at all possible to ensure a fluid user experience.
